changes/ee/feat-12517.en.md

```diff
@@ -0,0 +1,13 @@
+Implemented the Message Enrichment and Transformation feature.
+
+This allows users to transform incoming messages without the need to define SQL rules in the Rule Engine, just by using simple variform syntax.
+
+For example: if we want to take an incoming message encoded as Avro, decode it to JSON and then take the `tenant` client attribute from the publishing client and prefix it to the topic, before processing the result in Rule Engine, one would only need a transformation with the following configuration:
+
+```hocon
+payload_decoder = {type = avro, schema = myschema}
+payload_encoder = {type = json}
+operations = [
+  {key = "topic", value = "concat([client_attrs.tenant, '/', topic])"}
+]
```

Comment:
?
```suggestion
message_trasnformation {
    payload_decoder = {type = avro, schema = myschema}
    payload_encoder = {type = json}
    operations = [
        {key = "topic", value = "concat([client_attrs.tenant, '/', topic])"}
    ]
}
```

Reply:
Well, there are a few omitted, required fields.  I was trying to be succinct, since the full one would be larger.  :sweat_smile: 

Like:

```hcl
message_transformation {
  transformations = [
    {
      name = "mytransf"
      enable = true
      # ... even more fields ...
      payload_decoder = ...
      payload_encoder = ...
      operations = [
        ...
      ]
    }
  ]
}
```

Reply:
Should it be copy-and-pasteable?

Reply:
:heavy_check_mark: